opslevel/datasource_opslevel_property_definition.go

```diff
@@ -1,67 +1,118 @@
-// 			},
-// 		},
-// 	}
-// }
+// Ensure PropertyDefinitionDataSource implements DataSourceWithConfigure interface
+var _ datasource.DataSourceWithConfigure = &PropertyDefinitionDataSource{}
 
-// func datasourcePropertyDefinitionRead(d *schema.ResourceData, client *opslevel.Client) error {
-// 	identifier := d.Get("identifier").(string)
-// 	resource, err := client.GetPropertyDefinition(identifier)
-// 	if err != nil {
-// 		return err
-// 	}
+func NewPropertyDefinitionDataSource() datasource.DataSource {
+	return &PropertyDefinitionDataSource{}
+}
 
-// 	d.SetId(string(resource.Id))
+// PropertyDefinitionDataSource manages a PropertyDefinition data source.
+type PropertyDefinitionDataSource struct {
+	CommonDataSourceClient
+}
 
-// 	if err := d.Set("name", resource.Name); err != nil {
-// 		return err
-// 	}
-// 	if err := d.Set("description", resource.Description); err != nil {
-// 		return err
-// 	}
-// 	if err := d.Set("schema", resource.Schema.ToJSON()); err != nil {
-// 		return err
-// 	}
-// 	if err := d.Set("property_display_status", string(resource.PropertyDisplayStatus)); err != nil {
-// 		return err
-// 	}
+// PropertyDefinitionDataSourceModel describes the data source data model.
+type PropertyDefinitionDataSourceModel struct {
+	AllowedInConfigFiles  types.Bool   `tfsdk:"allowed_in_config_files"`
+	Description           types.String `tfsdk:"description"`
+	Id                    types.String `tfsdk:"id"`
+	Identifier            types.String `tfsdk:"identifier"`
+	Name                  types.String `tfsdk:"name"`
+	PropertyDisplayStatus types.String `tfsdk:"property_display_status"`
+	Schema                types.String `tfsdk:"schema"`
+}
 
-// 	return nil
-// }
+func NewPropertyDefinitionDataSourceModel(ctx context.Context, propertydefinition opslevel.PropertyDefinition, identifier string) PropertyDefinitionDataSourceModel {
+	return PropertyDefinitionDataSourceModel{
+		AllowedInConfigFiles:  types.BoolValue(propertydefinition.AllowedInConfigFiles),
+		Description:           types.StringValue(propertydefinition.Description),
+		Id:                    types.StringValue(string(propertydefinition.Id)),
+		Identifier:            types.StringValue(identifier),
+		Name:                  types.StringValue(propertydefinition.Name),
+		PropertyDisplayStatus: types.StringValue(string(propertydefinition.PropertyDisplayStatus)),
+		Schema:                types.StringValue(propertydefinition.Schema.ToJSON()),
+	}
+}
+
+func (d *PropertyDefinitionDataSource) Metadata(ctx context.Context, req datasource.MetadataRequest, resp *datasource.MetadataResponse) {
+	resp.TypeName = req.ProviderTypeName + "_property_definition"
+}
+
+func (d *PropertyDefinitionDataSource) Schema(ctx context.Context, req datasource.SchemaRequest, resp *datasource.SchemaResponse) {
+	resp.Schema = schema.Schema{
+		// This description is used by the documentation generator and the language server.
+		MarkdownDescription: "PropertyDefinition data source",
```

Comment:
```suggestion
		MarkdownDescription: "Data source for Property Definitions",
```

Suggestion to make this more human readable